client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,593 @@
+<template>
+    <div v-if="ready">
+        <h3>Share or Publish {{ modelClass }} `{{ item.title }}`</h3>
+        <div v-for="error in errors" :key="error">
+            <b-alert show variant="danger" dismissible @dismissed="errors = errors.filter((e) => e !== error)">
+                {{ error }}
+            </b-alert>
+        </div>
+        <br />
+        <div v-if="!hasUsername">
+            <div>To make a {{ modelClass }} accessible via link or publish it, you must create a public username:</div>
+            <form class="form-group" @submit.prevent="setUsername()">
+                <input class="form-control" type="text" v-model="newUsername" />
+            </form>
+            <b-button type="submit" variant="primary" @click="setUsername()">Set Username</b-button>
```

Comment:
This should be its own component, and I think we probably want to do some additional validation around the username. I haven't checked what we do for the registration, but I think we list what makes a  valid username there?

Maybe we already have a component for setting the username you could reuse ?
Maybe you could just redirect the user to the user preferences if that's too much work ?

Reply:
To be honest I never encountered this particular interface. This code wasn't written by us. It shows green, because we just copied this file in another directory.
Anyway, can you please give a hint, how do you create user without username? :sweat_smile: 
register page doesn't allow this, is it supposed to be done with a script?

Reply:
You can register a user via ldap, and then the user may not have a username. @nsoranzo @dannon do you remember the specifics here ? I vaguely remember we're now actually setting a username in all cases ?

If you want to test this out you can delete the username in the database for an existing user, I guess.

Reply:
Ok thanks, I managed to see it :+1: . I will create a separate PR for that, if it's fine by you.

Reply:
It is. I'll just run some tests now and probably merge later.